packages/atclient/include/atclient/atclient.h

```diff
@@ -69,13 +69,17 @@ int atclient_pkam_authenticate(atclient *ctx, const atclient_atkeys atkeys, cons
  * associated with your value.
  *
  * @param atclient the atclient context (must satisfy the two conditions stated above)
+ * @param root_conn initialized root connection
  * @param atkey the populated atkey to put the value into (must satisfy the two conditions stated above)
  * @param value the value to put into atServer
  * @param valuelen the length of the value (most of the time you will use strlen() on a null-terminated string for this
  * value)
  * @return int 0 on success
  */
-int atclient_put(const atclient *atclient, const atclient_atkey *atkey, const char *value, const size_t valuelen);
+int atclient_put(atclient *atclient, atclient_connection *root_conn, const atclient_atkey *atkey, const char *value,
```

Comment:
What does atclient_put need root_conn for?

Reply:
(Same for atclient_get_publickey, and atclient_get_sharedkey)

Reply:
If we want / need to lookup the other atSign's encryption keys, feel free to change this